### PR TITLE
Handle context file searching using forward slashes on Windows

### PR DESCRIPTION
**Pull request by DanTup**
Fixes https://github.com/sourcegraph/cody/issues/2198

## Test plan

- Run `pnpm test:e2e at-file` to run updated test, or:
- On Windows, open chat and try to attach a file in chat using backslashes and forward slashes for searching filenames

### CODE REVIEW

**Comment by kalanchan** [comment]
@dominiccooney another windows PR for ya :)

**Comment by DanTup** [comment]
I've opened https://github.com/sourcegraph/cody/pull/2241 too - not Windows specific, but depends on a function added in this PR to work. I'll rebase/merge that one after this lands (GitHub doesn't have a great way to do dependent PRs :-))

File: vscode/src/editor/utils/editor-context.ts [hunk 1]
```diff
@@ -46,6 +46,12 @@ export async function getFileContextFiles(
         return []
     }
 
+    // On Windows, if the user has typed forward slashes, map them to backslashes before
+    // running the search so they match the real paths.
+    if (path.sep === path.win32.sep) {
+        query = query.replaceAll(path.posix.sep, path.win32.sep)
+    }
```
Comment (dominiccooney) [inline vscode/src/editor/utils/editor-context.ts:53]:
```suggestion
    query = query.replaceAll(path.posix.sep, path.sep)
```

File: vscode/src/editor/utils/editor-context.ts [hunk 2]
```diff
@@ -175,6 +181,20 @@ function createContextFilePath(uri: vscode.Uri): ContextFile['path'] {
...
+ * Returna a relative path using the correct slash direction for the current platform.
+ */
+function asRelativePath(uri: vscode.Uri): string {
+    let relativePath = vscode.workspace.asRelativePath(uri.fsPath)
+    // asRelativePath returns forward slashes on Windows but we want to
+    // render a native path like VS Code does in the file quick-pick.
+    if (path.sep === path.win32.sep) {
+        relativePath = relativePath.replaceAll(path.posix.sep, path.win32.sep)
     }
+
```
Comment (dominiccooney) [inline vscode/src/editor/utils/editor-context.ts:198]:
```suggestion
    relativePath = relativePath.replaceAll(path.posix.sep, path.sep)

```

File: vscode/test/e2e/at-file-context-selecting.test.ts [hunk 3]
```diff
@@ -66,3 +84,7 @@ test('@-file empty state', async ({ page, sidebar }) => {
     await chatInput.press('Tab')
     await expect(chatInput).toHaveValue('@Main.java ')
 })
+
+function withPlatformSlashes(input: string) {
+    return path.sep === path.win32.sep ? input.replaceAll(path.posix.sep, path.win32.sep) : input
```
Comment (dominiccooney) [inline vscode/test/e2e/at-file-context-selecting.test.ts:89]:
```suggestion
    return input.replaceAll(path.posix.sep, path.sep)
```

**Review by dominiccooney — APPROVED** [review]
Beautiful.

Given your great comments I suggest we can simplify these conditionals to always replace / with the platform separator for a slight readability win.

If there were other Windows adaptations we were putting in those conditional blocks I would feel better about keeping the conditional blocks around.

**Pull request merged**